.github/workflows/_linux-test.yml

```diff
@@ -27,15 +27,46 @@ env:
   GIT_DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
 
 jobs:
+  filter:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@master
+        with:
+          fetch-depth: 1
+          submodules: false
+
+      - name: Parse ref
+        id: parse-ref
+        run: .github/scripts/parse_ref.py
+
+      - name: Select all requested test configurations
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        id: filter
+        run: |
+          .github/scripts/filter_test_configs.py \
+            --test-matrix "${{ inputs.test-matrix }}" \
+            --pr-number "${{ github.event.pull_request.number }}" \
+            --tag "${{ steps.parse-ref.outputs.tag }}"
+
+      - name: Print the filtered test matrix
+        run: |
+          echo "${{ steps.filter.outputs.test-matrix }}"
+
+    outputs:
+      test-matrix: ${{ steps.filter.outputs.test-matrix }}
+      is-test-matrix-empty: ${{ steps.filter.outputs.is-test-matrix-empty }}
```

Comment:
Is it by any chance possible to do something like `test-matrix.length` to see if the test matrix is empty?

I'm not sure what the limits of the language we get to use here is, though I vaguely remember Nikita saying any valid javascript will work.

Reply:
Well, `test-matrix` as a GHA output is always a string.  So it would looks like `{"include":[]}` when it's "empty" because it's the JSON dump of the test matrix data structure.  I could load the JSON, and check the length, but it seems to me that having another output variable is a cleaner way to check for this than making the workflow more complicated